addons/web/static/src/views/fields/selection/selection_field.js

```diff
@@ -86,14 +87,29 @@ export const selectionField = {
     component: SelectionField,
     displayName: _t("Selection"),
     supportedTypes: ["many2one", "selection"],
+    supportedOptions: [
+        {
+            label: _t("Autosave"),
+            name: "autosave",
+            type: "boolean",
+            default: true,
+            help: _t(
+                "If checked, the record will be saved immediately when the field is modified."
+            ),
+            viewTypes: ["kanban"],
+        },
+    ],
     isEmpty: (record, fieldName) => record.data[fieldName] === false,
-    extractProps({ attrs }, dynamicInfo) {
+    extractProps({ attrs, options, viewType }, dynamicInfo) {
         return {
+            autosave: viewType === "kanban" && "autosave" in options ? Boolean(options.autosave) : true,
             placeholder: attrs.placeholder,
             required: dynamicInfo.required,
             domain: dynamicInfo.domain(),
+            readonly: dynamicInfo.readonly,
```

Comment:
not a good idea, if i have a form view in readonly, my selection field shouldn't be in edit mode. Maybe ask to dbo, in kaban view, the selection shoud be in edit mode by default ?

Reply:
cc @bouvyd

Reply:
@FrancoisGe are you sure of what you're saying?

If I take the `boolean_toggle` as an example, we have something similar:
```js
export const booleanToggleField = {
    ...booleanField,
    component: BooleanToggleField,
    displayName: _t("Toggle"),
    supportedOptions: [
        {
            label: _t("Autosave"),
            name: "autosave",
            type: "boolean",
            default: true,
            help: _t(
                "If checked, the record will be saved immediately when the field is modified."
            ),
        },
    ],
    extractProps({ options }, dynamicInfo) {
        return {
            autosave: "autosave" in options ? Boolean(options.autosave) : true,
            readonly: dynamicInfo.readonly,
        };
    },
};
```
and yet in a form view with `edit="0"`, the widget is properly readonly.

Similarly in this branch, if I try that with a selection widget it is properly readonly.

Reply:
@bouvyd Yes, you're right, I didn't give you the right use case.  In a form view, if you have an x2many in list view mode with a selection field. I think that all the selection fields will be in edit mode (if you have widget="selection" on the <field>). We have a BooleanToggleField extension for the list view.